### ISSUE TITLE
byuu.org/emulation/higan/ leads to 404

### ISSUE THREAD

**Issue by Marat-Tanalin** [byuu/byuu.org#6]
https://byuu.org/emulation/higan/ is one of the URLs that higan was [available](https://news.ycombinator.com/item?id=18721934) at in 2018 or so (the original site author did change URLs multiple times).

A simplest redirect can be done with a `refresh` meta element, so in the simplest case the `index.htm` page in the `/emulation/higan/` folder would contain e.g. the following:

    <!DOCTYPE html>
    <meta http-equiv="refresh" content="0; url=/higan/" />

**Comment by Screwtapello** [comment]
I'm not sure I *can* add a redirect to that URL, since the site is now built from Markdown files by GitHub Pages, rather than actual raw HTML.

If there's a lot of broken links to that URL, it's probably worth spending time figuring out how to set up a redirect, but I can't seem to convince Google or DDG to tell me if that's true. Is the lack of a redirect causing a problem for you personally, or is it just something you noticed while reading an old HN article?

**Comment by Marat-Tanalin** [comment]
That's a link I have in my bookmarks and browser history. The HN thread is just an example of that link in the wild. Not a big problem, just letting you know.

I've tested on my GitHub.io [site](marat-tanalin.github.io) (just created for testing purposes), and it’s apparently possible to use HTML instead of Markdown on GitHub.io. See the [`index.htm`](https://github.com/Marat-Tanalin/marat-tanalin.github.io/blob/master/index.htm) file in the root of my test repo, as an&nbsp;example.

**Comment by Screwtapello** [comment]
I dug around and found a list of historical redirections that byuu had set up. Some of them redirected now-defunct pages to other now-defunct pages, but a few of them redirected to currently-existing pages, including `/emulation/higan/` → `/higan/`, so I [created](https://github.com/byuu/byuu.org/commit/4568e70bebd5e2c953e14fa61aef8e75cd8dc4a6) static HTML redirections for all of them.

Unfortunately, although the redirects are [clearly in the repo](https://github.com/byuu/byuu.org/blob/4568e70bebd5e2c953e14fa61aef8e75cd8dc4a6/emulation/higan/index.html), https://byuu.org/emulation/higan/ still gives me a 404. I'm not sure what's going on here.

**Comment by Screwtapello** [comment]
On the other hand, a bunch of people in the byuucord report the redirect works fine for them. I'm not sure what's going on.

**Comment by Marat-Tanalin** [comment]
Thanks, the redirects work fine for me.

As for why the result is not visible immediately: there is a delay between changing a file in the repo and reflecting the changes on the corresponding GitHub.io webpage. This is apparently caused by server-side caching (caching is used to improve performance). I encountered the same with my test pages on GitHub.io, but I was able to see the updated file in a few minutes. There is also browser-side (client-side) caching, so it might be needed to do a forced refresh by pressing `F5` or (more reliable) `Ctrl+F5` in the web browser even if the server-side cache is already cleared and server already outputs the updated version to those not visited the page before.

**Comment by Marat-Tanalin** [comment]
For what it’s worth, there is no need for including the host into the URL if the destination webpage is in the same domain.

So instead of:

    <!DOCTYPE html>
    <meta http-equiv="refresh" content="0; url=//byuu.org/higan/" />

it would be enough to use:

    <!DOCTYPE html>
    <meta http-equiv="refresh" content="0; url=/higan/" />

**Comment by Screwtapello** [comment]
I tested in three different browsers, plus Firefox's Private Browsing mode and with force-reload, and the redirect failed to show up. As the timestamps show, I pushed the change at 16:15, I was still having problems at 16:30 and (I think) even by 17:08.

Nevertheless, it's fixed for me now too, so :shrug: 

> For what it’s worth, there is no need for including the host into the URL...

Yeah, but the historical redirection data I had included the hostname, and I used a one-line shell-script to convert it into a bunch of `index.html` files; it would have been more work to remove the host info. :)
